ros/ros/ros2.js

```diff
@@ -133,52 +142,89 @@ class ROS2 {
     return list.find(({name}) => name == service)?.types[0];
   }
 
+  _destroySubscriber(subscriber){
+    if (subscriber && !subscriber.__destroyed){
+      subscriber.__destroyed = true;
+      this.node.destroySubscription(subscriber);
+    }
+  }
   /** Subscribe to the named topic of the named type. Each time a new message
   * is received the provided callback is called. For available options see
   * https://robotwebtools.github.io/rclnodejs/docs/0.22.3/Node.html#createSubscription.
   * The default `options.qos.reliability` is best-effort.
+  * options object can contain: "throttleMs": throttle-in-milliseconds.
   * */
   subscribe(topic, type, onMessage, options = {}) {
     this.requireInit();
+    let firstLatchedMessage;
     const ros2Type = toROS2Type(type);
-    const sub = this.node.createSubscription(
-      ros2Type, topic, {qos, ...options}, onMessage);
-    this.subscriptions[topic] = sub;
+    const _onMessage = options?.throttleMs ?
+      _.throttle(onMessage, options.throttleMs) :
+      onMessage;
+
+    const latchingSub = this.node.createSubscription(
+      ros2Type, topic, {qos: latchingQos, ...options}, (msg) => {
+        this._destroySubscriber(latchingSub);
+        firstLatchedMessage = msg;
+        _onMessage(msg);
+      }
+    );
+
+    const volatileSub = this.node.createSubscription(
+      ros2Type, topic, {qos: volatileQos, ...options}, (msg) => {
+        this._destroySubscriber(latchingSub);
+        if (firstLatchedMessage){
+          if(_.isEqual(firstLatchedMessage, msg)){
```

Comment:
nit: `if(_.isEqual` -> `if (_.isEqual` (here and elsewhere); same at the end: `){` -> `) {`.